types/src/v0/impls/header.rs

```diff
@@ -1099,6 +1100,36 @@ impl BlockHeader<SeqTypes> for Header {
             )?,
         })
     }
+
+    fn auth_root(&self) -> anyhow::Result<Option<[u8; 32]>> {
+        match self {
+            Header::V1(_) | Header::V2(_) | Header::V3(_) => Ok(None),
+            Header::V4(header) => {
+                let mut reward_root_bytes = Vec::new();
+                let el_root = [0; 32];
+                let placeholder_root = [0; 32];
+                header
+                    .reward_merkle_tree_root
+                    .serialize_compressed(&mut reward_root_bytes)?;
+                // The number of reward root bytes produced by `serialize_compressed()` is 48.
+                ensure!(
+                    reward_root_bytes.len() == 48,
+                    "unexpected reward root byte length: expected 48, got {}",
+                    reward_root_bytes.len()
+                );
+
+                let mut hasher = Keccak256::new();
+
+                hasher.update(reward_root_bytes);
+                hasher.update(el_root);
+                hasher.update(placeholder_root);
```

Comment:
@imabdulbasit @alxiong @mrain I'm thinking we should add a few more placeholders to make it easier to add more authenticated data in the future.  And let's just call them all placeholder or "reserved" for now because we don't know exactly if the first thing we add will be EL or something else.